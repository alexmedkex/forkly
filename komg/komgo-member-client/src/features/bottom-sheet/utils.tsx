import { BottomSheetItem, BottomSheetStatus, BottomSheetAction, BottomSheetItemType } from './store/types'
import React from 'react'

export const generateBottomsheetTitle = (items: BottomSheetItem[]) => {
  let textHeader = ''
  if (items[0]) {
    const latestStatus = items[0].state
    const count = items.filter(x => x.state === latestStatus).length
    textHeader = getPrintableStatus(latestStatus, count)
  }
  return textHeader
}

const getPrintableStatus = (latestStatus: BottomSheetStatus, count: number) => {
  switch (latestStatus) {
    case BottomSheetStatus.REGISTERED:
      return `${count} document${count > 1 ? 's' : ''} registered`
    case BottomSheetStatus.PENDING:
      return `${count} document${count > 1 ? 's' : ''} pending`
    case BottomSheetStatus.FAILED:
      return `${count} failed registration`
  }
}

export const getStatus = (action: BottomSheetAction): BottomSheetStatus => {
  return BottomSheetStatus.PENDING
}

export function fakeBottomsheetItem<T extends BottomSheetItem>(partial?: Partial<T>): BottomSheetItem {
  const { id, state, name, itemType } = {
    ...{
      id: '-1',
      name: 'anon',
      state: BottomSheetStatus.PENDING,
      itemType: BottomSheetItemType.REGISTER_KYC_DOCUMENT
    },
    ...partial
  }
  return { id, state, name, itemType }
}
