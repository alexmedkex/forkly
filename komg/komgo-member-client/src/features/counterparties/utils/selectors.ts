import { TaskWithUser } from '../../tasks/store/types'
import * as _ from 'lodash'
import { Counterparty } from '../store/types'
import { IMember } from '../../members/store/types'

export const findTaskByCounterpartyStaticId = (
  tasks: TaskWithUser[] = [],
  counterpartyStaticId: string | undefined | number
): TaskWithUser | undefined => {
  return _.find(tasks, task => task.task.counterpartyStaticId === counterpartyStaticId)
}

export const getCompanyName = (company: IMember | Counterparty, defaultName: string = '-') => {
  if (company && company.x500Name && company.x500Name.CN) {
    return company.x500Name.CN
  }
  return defaultName
}
