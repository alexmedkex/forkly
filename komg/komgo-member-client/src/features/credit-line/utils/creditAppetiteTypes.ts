import { CreditLineType } from '../store/types'
import { Products } from '../../document-management/constants/Products'
import { SubProducts } from '../../document-management/constants/SubProducts'

export const findFeature = (data: any): CreditLineType | null => {
  const checkingObj = data.context || data
  if (
    checkingObj.productId === Products.TradeFinance &&
    checkingObj.subProductId === SubProducts.ReceivableDiscounting
  ) {
    return CreditLineType.RiskCover
  }
  if (checkingObj.productId === Products.TradeFinance && checkingObj.subProductId === SubProducts.LetterOfCredit) {
    return CreditLineType.BankLine
  }
  return null
}
