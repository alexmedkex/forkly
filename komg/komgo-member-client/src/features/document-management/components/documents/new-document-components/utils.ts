import { HasId, HasName } from '../../../store/types'

export const hasIdAndNameToOption = (collection: Array<HasId & HasName>): Array<{ key: string }> => {
  const originalCollection = collection.map(el => {
    return { key: el.id, text: el.name, value: el.id }
  })
  return originalCollection.reduce((unique: Array<{ key: string }>, o: { key: string }) => {
    if (!unique.some((obj: { key: string }) => obj.key === o.key)) {
      unique.push(o)
    }
    return unique
  }, [])
}
