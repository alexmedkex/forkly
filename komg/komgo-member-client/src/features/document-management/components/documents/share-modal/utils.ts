import { ICompanyTableItemWithSharedDocs, ICounterpartyWithDocuments } from './SelectedCounterparties'
import { FlagNameValues } from 'semantic-ui-react'
import * as i18nIsoCountries from 'i18n-iso-countries'

export function pluralize(word: string, count: number, singleSuffix: string, pluralSuffix: string) {
  if (count > 1) {
    return `${word}${pluralSuffix}`
  }

  return `${word}${singleSuffix}`
}

export const buildCounterpartyWithDocumentsPickerItems = (
  counterpartiesWithDocs: ICounterpartyWithDocuments[]
): ICompanyTableItemWithSharedDocs[] => {
  i18nIsoCountries.registerLocale(require('i18n-iso-countries/langs/en.json'))
  return counterpartiesWithDocs.map(item => ({
    name: item.counterparty.x500Name.CN,
    countryName: i18nIsoCountries.getName(item.counterparty.x500Name.C.trim(), 'en'),
    country: item.counterparty.x500Name.C.toLocaleLowerCase().trim() as FlagNameValues,
    location: item.counterparty.x500Name.L,
    id: item.counterparty.staticId,
    sharedDocuments: item.documents
  }))
}
