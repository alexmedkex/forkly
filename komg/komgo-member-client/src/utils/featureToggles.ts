export enum FeatureToggle {
  Default = 'default',
  NewRequestDocuments = 'newRequestDocuments',
  CompanyDeactivation = 'deactivation'
}

const FEATURE_PARAM_NAME = 'ftoggles'

export const isFeatureToggleEnabled = (featureToggle: FeatureToggle, queryString: string): boolean => {
  const features = getEnabledFeatureToggles(queryString)
  return features.includes(featureToggle)
}

export const getEnabledFeatureToggles = (queryString: string): string[] => {
  let features = []
  // We must use string interpolation, otherwise webpack will put a result of the expression, i.e. false
  // That's because string '%REACT_APP_ALLOW_FEATURE_TOGGLES%' !== 'true' (see Dockerfile)
  const allowFeatureToggles = `${process.env.REACT_APP_ALLOW_FEATURE_TOGGLES}`

  if (allowFeatureToggles === 'true') {
    if (process.env.REACT_APP_ENABLED_FEATURE_TOGGLES) {
      const envFeatures = process.env.REACT_APP_ENABLED_FEATURE_TOGGLES.split(',').map(f => f.trim())
      features = features.concat(envFeatures)
    }

    const urlSearchParams = new URLSearchParams(queryString)
    if (urlSearchParams.has(FEATURE_PARAM_NAME)) {
      const queryFeatures = urlSearchParams
        .get(FEATURE_PARAM_NAME)
        .split(',')
        .map(f => f.trim())
      features = features.concat(queryFeatures)
    }
  }
  return features
}
