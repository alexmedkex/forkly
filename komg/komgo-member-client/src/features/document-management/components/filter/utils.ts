import { Category, DocumentType } from '../../index'
import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'
import { categoryToColor } from '../documents/document-library/categoryToColor'

export const getDocumentTypeFilterOptions = (categories: Category[], types: DocumentType[]) => {
  return sortCategories(categories).map(c => ({
    name: c.name,
    value: c.id,
    display: {
      color: categoryToColor[c.id]
    },
    items: sortDocumentTypes(types)
      .filter(t => t.category.id === c.id)
      .map(t => ({
        name: t.name,
        value: t.id
      }))
  }))
}
