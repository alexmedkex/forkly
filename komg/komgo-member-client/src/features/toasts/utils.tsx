import { css } from 'glamor'
import React from 'react'
import { toast } from 'react-toastify'
import { ToastContainerIds, ToastIds } from '../../utils/toast'
import Toast, { ToastAnimation } from './containers/Toast'

export enum TOAST_TYPE {
  Ok = 0,
  Error = 1
}

export const displayToast = (msg: string, status: TOAST_TYPE = TOAST_TYPE.Ok) => {
  const closeTimeoutMs = 5000
  if (toast.isActive(ToastIds.CustomToast)) {
    toast.update(ToastIds.CustomToast, {
      render: <Toast text={msg} closeToast={() => undefined} />,
      autoClose: closeTimeoutMs,
      containerId: ToastContainerIds.Custom
    })
  } else {
    toast(<Toast text={msg} closeToast={() => undefined} isError={!!status.valueOf()} />, {
      className: css({
        background: '#1c2936',
        padding: '0px !important',
        minHeight: '0px !important',
        marginBottom: 'unset !important'
      }),
      position: toast.POSITION.BOTTOM_LEFT,
      hideProgressBar: true,
      autoClose: closeTimeoutMs,
      transition: ToastAnimation,
      closeOnClick: false,
      draggable: false,
      toastId: ToastIds.CustomToast,
      containerId: ToastContainerIds.Custom
    })
  }
}
