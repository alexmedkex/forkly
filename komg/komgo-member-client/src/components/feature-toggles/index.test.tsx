import * as React from 'react'
import * as renderer from 'react-test-renderer'

import { Feature, FeatureProvider } from '.'
import { FeatureToggle } from '../../utils/featureToggles'

describe('FeatureEnabler', () => {
  it('should not show feature if is not included', () => {
    const features = []
    expect(
      renderer
        .create(
          <FeatureProvider features={features}>
            <Feature featureToggle={FeatureToggle.Default}>
              <p>test1</p>
            </Feature>
          </FeatureProvider>
        )
        .toJSON()
    ).toMatchSnapshot()
  })

  it('should show feature if is included', () => {
    const features = [FeatureToggle.Default]
    expect(
      renderer
        .create(
          <FeatureProvider features={features}>
            <Feature featureToggle={FeatureToggle.Default}>
              <p>test1</p>
            </Feature>
          </FeatureProvider>
        )
        .toJSON()
    ).toMatchSnapshot()
  })

  it('should not show feature if is not included but contain different ones', () => {
    const features = ['dummyFt']
    expect(
      renderer
        .create(
          <FeatureProvider features={features}>
            <Feature featureToggle={FeatureToggle.Default}>
              <p>test1</p>
            </Feature>
          </FeatureProvider>
        )
        .toJSON()
    ).toMatchSnapshot()
  })

  it('should show inactive component if FeatureToggle is not included', () => {
    const features = []
    expect(
      renderer
        .create(
          <FeatureProvider features={features}>
            <Feature featureToggle={FeatureToggle.Default} inactiveComponent={() => <p>testInactive</p>}>
              <p>test1</p>
            </Feature>
          </FeatureProvider>
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
