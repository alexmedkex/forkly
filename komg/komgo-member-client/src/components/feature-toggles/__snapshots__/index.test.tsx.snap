// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureEnabler should not show feature if is not included 1`] = `null`;

exports[`FeatureEnabler should not show feature if is not included but contain different ones 1`] = `null`;

exports[`FeatureEnabler should show feature if is included 1`] = `
<p>
  test1
</p>
`;

exports[`FeatureEnabler should show inactive component if FeatureToggle is not included 1`] = `
<p>
  testInactive
</p>
`;
